pkg/v4/resource/chartconfig/desired.go

```diff
@@ -45,6 +45,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		desiredChartConfigs = append(desiredChartConfigs, chartConfig)
 	}
 
+	// Only enable External DNS for Azure.
+	if r.provider == label.ProviderAzure {
+		chartConfig := newExternalDNSChartConfig(clusterConfig, r.projectName)
+		desiredChartConfigs = append(desiredChartConfigs, chartConfig)
+	}
```

Comment:
This is the wrong place to do this IMO. The resources in `pkg/` are actually meant for all providers. There are specific resources for Azure and we should deal with Azure specifics there.

Reply:
Sorry, i'm not into cluster-operator. May be @rossf7 can answer, why we have this under `pkg`?

Reply:
The logic for creating chartconfigs is the same for all providers. But we need to be able to install provider specific charts like in this case.

Other way to do this is to extract all logic to a service and have resources per provider with different desired states. That is doable but some code duplication. WDYT?

cc @fgimenez @indyfree